lib/Utils/Backends.pm

```diff
@@ -288,4 +289,58 @@ sub is_ssh_installation {
           (($videomode =~ /text/) && (is_pvm || is_s390x)));
 }
 
+=head2 update_ssh_console_netconfig
+
+Update the network configuration using current netconfig settings. This will help
+refresh file like /etc/resolv.conf, so associated machine will get the latest or
+corrected name servers to continue to function as normal, for example, establish 
+connection back to worker machine by using FQDN or communicate with system outside 
+using FQDN. If "netconfig update -f" can not help, adding functioning name servers
+into /etc/resolv.conf provided by worker machine. Record failure and even terminate 
+test run if host fails to communicate outside at the last. Arguments are address 
+of the system on which netconfig will operate and the number of retries.
+
+=cut
+
+sub update_ssh_console_netconfig {
+    my (%args) = @_;
+    $args{dest} //= 'localhost';
+    $args{retry} //= 3;
+
+    my $pingaddr = version_utils::is_opensuse() ? 'openqa.opensuse.org' : 'openqa.suse.de';
+    my $sshcmd = $args{dest} eq 'localhost' ? "" : "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root\@$args{dest}";
+    my $pingret = script_run("$sshcmd timeout --kill-after=1 --signal=9 20 ping -c5 $pingaddr");
+    while ($args{retry} > 0 and $pingret != 0) {
+        script_run("$sshcmd timeout --kill-after=1 --signal=9 20 netconfig update -f");
+        $pingret = script_run("$sshcmd timeout --kill-after=1 --signal=9 20 ping -c5 $pingaddr");
+        $args{retry} -= 1;
+    }
+    save_screenshot;
+    if ($pingret != 0) {
+        my $resolv_file = "/etc/resolv.conf";
+        assert_script_run("$sshcmd sed -i -r \'/^nameserver.*\$/d\' $resolv_file") if (script_run("$sshcmd cat $resolv_file | grep \"^nameserver.*\$\"") == 0);
+        my $nameserver = "";
+        open my $fh, '<', $resolv_file or die "Unable to open $resolv_file for input";
+        while (my $line = <$fh>) {
+            if ($line =~ /^nameserver.*$/) {
+                my $temp = $line;
+                $temp =~ s/\n//g;
+                $nameserver .= "$temp\\n";
+            }
+        }
```

Comment:
With a little less code:
```perl
for my $line (split "\n", path($resolve_file)->slurp) {
  next unless $line =~ /^nameserver/;
  ...
}
```

Reply:
Or as a one-liner, but i find this less readable:
```perl
my $nameservers = join '', map { "$_\\n" } grep { /^nameserver/ } split "\n", path($resolve_file)->slurp;
```

Reply:
Done. Use this one-liner.